source/documentation/deploying_apps/deploying_java.md

```diff
@@ -0,0 +1,122 @@
+    import javax.servlet.ServletException;
+    import javax.servlet.http.HttpServletRequest;
+    import javax.servlet.http.HttpServletResponse;
+    import org.eclipse.jetty.server.Request;
+    import org.eclipse.jetty.server.Server;
+    import org.eclipse.jetty.server.handler.AbstractHandler;
+
+
+    public class App extends AbstractHandler
+    {
+        @Override
+        public void handle( String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response ) throws IOException, ServletException
+        {
+               response.setContentType("text/html; charset=utf-8");
+               response.setStatus(HttpServletResponse.SC_OK);
+               response.getWriter().println("<h1>Hello World</h1>");
+               baseRequest.setHandled(true);
+        }
+
+        public static void main( String[] args ) throws Exception
+        {
+                int port = Integer.parseInt(System.getenv("PORT"));
+                Server server = new Server(port);
+                server.setHandler(new App());
+
+                server.start();
+                server.join();
+        }
+    }
+    ```
+4. Compile your class:
+
+    ```bash
+    javac -d . -cp ./jetty.jar App.java
+    ```
+
+5. Combine your class with Jetty to create an executable jar file:
+
+    ```bash
+    jar uvfe ./jetty.jar app.App app/App.class
+    ``` 
+
+6. Push your jar file.
+
+    ```bash
+    cf push your-app-name -b java_buildpack -p ./jetty.jar
+    ```
+
+You have now deployed your Java application. Your application should now be accessible over HTTPS from the URL provided in the output.
+
+If your application requires arguments in order to start, you can set the `JBP_CONFIG_JAVA_MAIN` environment variable. For example to add `server` and `config.yml` as arguments for your app, you could add the following to your application's `manifest.yml`:
+
+```yaml
+env:
+  JBP_CONFIG_JAVA_MAIN: '{ arguments: "server config.yml" }
+```
+
+### Deploying a WAR file
+
+If your application can be packaged as a `.war` file you can deploy with the following:
+
+```bash
+cf push your-app-name -b java_buildpack -p your-app.war
+```
+
+If you need to use a specific version of Tomcat you can set the environment variable `JBP_CONFIG_TOMCAT`. For example to use Tomcat 8 you could add the following to your application's `manifest.yml`:
+
+```yaml
+env:
+  JBP_CONFIG_TOMCAT: '{ tomcat: { version: 8.0.+ } }'
+```
+Note that you do **not** need to deploy Tomcat along with your application. The Java buildpack will run the servlet 2 and 3 web applications.
+
+For more configuration options see [Tomcat Container](https://github.com/cloudfoundry/java-buildpack/blob/master/docs/container-tomcat.md) [external link].
+
+### Deploying other JVM-based applications
+
+The Java buildpack supports running any JVM-based applications with little or zero additional configuration. For detailed examples of deploying applications using other frameworks and JVM-based languages:
```

Comment:
Tom's comment on the pivotal story says "We also need to include that the Java buildpack supports "JVM-based languages" such as Scala, Clojure etc".

I think it's worth mentioning the common ones we know about by name so that people searching for eg Scala will find this section.

Reply:
Do we need to include links to Scala and Clojure docs? If we mention them in the text but don't include links in the list, it might look a bit strange.

Reply:
I think mentioning them by name is probably enough.

Maybe use the line from Tom's comment eg "The Java buildpack supports running applications using JVM-based languages (such as Scala, Clojure etc) with little..."

Reply:
Ok, I've changed the text accordingly